### PR TITLE
HTTP client transport

### PR DESCRIPTION
**Pull request by dmytro-lazebnyi**

### CODE REVIEW

File: ClientSdk/src/main/java/io/scalecube/gateway/clientsdk/http/HttpClientTransport.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    throw new UnsupportedOperationException("Request stream is not supported by HTTP/1.x");
+  }
+
+  @Override
+  public Mono<Void> close() {
+    return Mono.empty();
+  }
+
+  private boolean isError(int httpCode) {
+    int httpCodeType = httpCode / 100;
```
Comment (segabriel) [inline ClientSdk/src/main/java/io/scalecube/gateway/clientsdk/http/HttpClientTransport.java]: why not is it just like `return httpCode >= 400` ?

File: Examples/src/main/java/io/scalecube/gateway/examples/GreetingServiceCancelCallback.java [hunk 2]
```diff
@@ -1,74 +0,0 @@
-package io.scalecube.gateway.examples;
-
-import reactor.core.publisher.Flux;
-import reactor.core.publisher.Mono;
-
-public class GreetingServiceCancelCallback implements GreetingService {
```
Comment (segabriel) [inline Examples/src/main/java/io/scalecube/gateway/examples/GreetingServiceCancelCallback.java]: Why did you remove it? How will we check case when a client will be disconnected or will send the completion signal? How will we sure that our service canceled work inside after a client close its connection?

**Comment by dmytro-lazebnyi** [comment]
@segabriel I reverted deleting of `GreetingServiceCancelCallback`

File: ClientSdk/src/main/java/io/scalecube/gateway/clientsdk/http/HttpClientTransport.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        () ->
+            httpClient
+                .post(
+                    request.qualifier(),
+                    httpRequest -> {
+                      LOGGER.info("Sending request {}", request);
+
+                      return httpRequest
+                          .failOnClientError(false)
+                          .failOnServerError(false)
```
Comment (segabriel) [inline ClientSdk/src/main/java/io/scalecube/gateway/clientsdk/http/HttpClientTransport.java]: :+1:
